### PR TITLE
mpy-cross/mpy_cross: Add a way to query the mpy version.

### PR DESCRIPTION
**Pull request by jimmo**
This returns the:
 - mpy_version
 - mpy_sub_version
 - native_arch
for files compiled with this mpy-cross binary.

I considered doing this by adding a different `--version` mode to mpy-cross and parsing that. Happy to do it either way (and open to suggestions!)

Also fixes some other issues in the mpy_cross Python wrapper, mostly the default path (all current uses specify the path explicitly though so that's why this wasn't a problem).

_This work was funded through GitHub Sponsors._

### CODE REVIEW

File: mpy-cross/mpy_cross/__init__.py [hunk 1]
```diff
@@ -54,17 +55,40 @@
...
-__all__ = ["compile", "run", "CrossCompileError"]
+__all__ = ["version", "compile", "run", "CrossCompileError"]
 
 
 class CrossCompileError(Exception):
     pass
 
 
-def find_mpy_cross_binary(mpy_cross):
+def version(march=None, mpy_cross=None):
```
Comment (dlech) [inline mpy-cross/mpy_cross/__init__.py:65]: Might be better to call this `abi_version` to not confuse it with the mpy-cross/MicroPython version.

Comment (dlech) [reply mpy-cross/mpy_cross/__init__.py:65]: Also, a doc string explaining the parameters and return value would be helpful.

**Review by dlech — COMMENTED** [review]
What is the use case for including the arch in the version?

**Comment by jimmo** [comment]
> What is the use case for including the arch in the version?

It's essentially a (complicated) way to map possible values of the `-march` flag to the resulting feature byte value.

**Comment by jimmo** [comment]
Thanks @dlech 

Updated to rename the new function to `mpy_version()` (we use "ABI" informally, but most things refer to this as the "mpy version"), and to parse the output of `--version` instead.

Also added a fix for the native arch constants (adding them to `__all__`), and docstrings.

**Pull request merged**